app/services/git/syncs_updated_tracks.rb

```diff
@@ -0,0 +1,44 @@
+class Git::SyncsUpdatedTracks
+  QUIET_PERIOD = 10.seconds
+
+  def self.run
+    new.run
+  end
+
+  def self.sync
+    new.sync
+  end
+
+  def run
+    loop do
+      sync
+      puts "Sleeping"
+      sleep QUIET_PERIOD
+    end
+  end
+
+  def sync
+    find_track_update
+    if track_update
+      puts "Syncing outstanding tracks"
+      sync_track_update
+    end
+  end
+
+  private
+  attr_reader :track_update
+
+  def find_track_update
+    @track_update = TrackUpdate.
+      joins(:track_update_fetches).
+      where(synced_at: nil).
+      where.not(track_update_fetches: { completed_at: nil }).
+      group("track_update_fetches.id").
+      having("count(track_update_id) = #{ENV["NUM_WEBSERVERS"]}").
```

Comment:
Nice use of `having` there, I still find active record queries hard to do. As I mentioned in the top-level comment. I wonder if we should externalise this in a YAML file so it can be set on a per-env basis. That's the pattern we've been following throughout this project, but happy to be overruled if the env var has advantages. See https://github.com/exercism/prototype/blob/master/app/models/git/github_profile.rb#L4 and https://github.com/exercism/prototype/blob/master/config/auth.yml for an example. It might be we want to refactor some of these configs and have one for managing the git integration stuff, maybe...

Reply:
Sure, I don't know any advantages of having an ENV var over a YAML file. I'll be glad to change this to use the YAML file.